tests/unit/machine/IterativeMachine_unittest.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ */
+
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <rxcpp/rx-lite.hpp>
+#include <shogun/lib/Signal.h>
+
+#include "environments/LinearTestEnvironment.h"
+#include "utils/SGObjectIterator.h"
+#include <shogun/base/some.h>
+#include <shogun/classifier/AveragedPerceptron.h>
+#include <shogun/classifier/Perceptron.h>
+#include <shogun/classifier/svm/NewtonSVM.h>
+#include <shogun/machine/IterativeMachine.h>
+#include <shogun/machine/LinearMachine.h>
+
+using namespace shogun;
+
+std::set<std::string> sg_linear_machines = {"Perceptron", "AveragedPerceptron",
```

Comment:
ok! This is definitely better than before. I have a question: If a test fails, is it easy to see which class is the offender?

Filtering the tests obviously is not possible via this though. Compile time is an advantage to the typed version of this we had before (we define a bunch of types and then essentially template the test). This then allows for filtering. Maybe in a follow up PR, we can do something about that. I think the current blocker was that it is hard to find all subclasses automatically. However, if we simply define a type list (as you have done here with strings), that is also acceptable for now.

@gf712 might also be interesting for you

Reply:
It should be easy using `SCOPED_TRACE`. 
Hard coding the linear machines may not be so scaleable. Can't the `class_list` script be extended to detect hierarchies?
I don't get your point about filtering, and why templated tests can be better.

Reply:
Cool add the scoping!
Yes we can do this extension, but for now the explicit list is fine.
Compiled types can be helpful but for now this is good!